### PR TITLE
Fix added pep8 conventions

### PR DESCRIPTION
**Pull request by bwrsandman**
Put boolean operators at end of lines instead of beggining of newlines
Put lambda definition in fields.function def
Convert lambda into def

### CODE REVIEW

File: base_partner_merge/base_partner_merge.py [hunk 1]
```diff
@@ -381,8 +381,10 @@ def _merge(self, cr, uid, partner_ids, dst_partner=None, context=None):
...
         _logger.info("dst_partner: %s", dst_partner.id)
 
-        call_it = lambda function: function(cr, uid, src_partners,
-                                            dst_partner, context=context)
+        def call_it(function):
+            return function(
+                cr, uid, src_partners, dst_partner, context=context
+            )
 
         call_it(self._update_foreign_keys)
```
Comment (pedrobaeza) [inline base_partner_merge/base_partner_merge.py]: You can do directly:

```
self._update_foreign_keys(cr, uid, src_partners, dst_partner, context=context)
self._update_reference_fields(cr, uid, src_partners, dst_partner, context=context)
```

File: partner_firstname/partner.py [hunk 2]
```diff
@@ -68,10 +68,10 @@ def _write_name(
         In addition an heuristic avoids to keep a firstname without a
         non-blank lastname
         """
-        field_value = (field_value
-                       and not field_value.isspace()
-                       and field_value
-                       or False)
+        field_value = (
+            field_value and not field_value.isspace() and field_value or
+            False
```
Comment (yvaucher) [inline partner_firstname/partner.py]: Could you write?

```
field_value if field_value and not field_value.isspace() else False
```

File: partner_relations/model/res_partner_relation.py [hunk 3]
```diff
@@ -87,8 +87,8 @@ def get_values(this, dummy_field_names, dummy_arg, context=None):
                 ((this.type_id.id) * 10) + (on_right_partner and 1 or 0))
             # partner_id_display
             values['partner_id_display'] = (
-                on_right_partner and this.left_partner_id.id
-                or this.right_partner_id.id
+                on_right_partner and this.left_partner_id.id or
+                this.right_partner_id.id
```
Comment (yvaucher) [inline partner_relations/model/res_partner_relation.py]: We can use `if else` here

**Comment by yvaucher** [comment]
Just 2 little non blocking remarks otherwise :+1:

**Comment by hbrunn** [comment]
`./partner_relations/model/res_partner_relation.py:104:17: F812 list comprehension redefines 'self' from line 76`

I think you shouldn't touch the list comprehension in the last file. And what's wrong with `this` style-wise anyway?

**Comment by bwrsandman** [comment]
It's defined in [PEP 8](https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments):

> ## Function and method arguments 
> 
>  Always use `self` for the first argument to instance methods.
> 
> Always use `cls` for the first argument to class methods.
> 
> If a function argument's name clashes with a reserved keyword, it is generally better to append a single trailing underscore rather than use an abbreviation or spelling corruption. Thus `class_` is better than `clss` . (Perhaps better is to avoid such clashes by using a synonym.)

File: base_partner_merge/base_partner_merge.py [hunk 4]
```diff
@@ -381,12 +381,15 @@ def _merge(self, cr, uid, partner_ids, dst_partner=None, context=None):
...
 
-        call_it = lambda function: function(cr, uid, src_partners,
-                                            dst_partner, context=context)
-
-        call_it(self._update_foreign_keys)
-        call_it(self._update_reference_fields)
-        call_it(self._update_values)
+        self._update_foreign_keys(
+            cr, uid, src_partners, dst_partner, context=context
+        )
```
Comment (lmignon) [inline base_partner_merge/base_partner_merge.py:386]: Why to you put the last ')' on a new line? Sometimes it's at the end of  line sometimes on a new line...  (I know it's a kind a nit-picking :smirk: )

**Comment by lmignon** [comment]
:+1:  (code review)

File: base_partner_merge/base_partner_merge.py (hunk 4 above)
Comment (bwrsandman) [inline base_partner_merge/base_partner_merge.py:386]: The reason I do it is for symmetry and simpler future diffs.

If the `(` has nothing to the right of it, then I put nothing at the left of `)`
It also saves a character on the argument line.

Additionally, if the argument list got longer than 80 chars, I would do two things, put a comma at the end of the last arg and put the closing parens on its own line:

``` python
foo(
    arg1,
    arg2,
    arg3,
    arg4,
    ...
    argN,
)
```

Notice the comma after `argN`, this means that if `argN+1,` is added, the diff only has one line.

``` diff
 foo(
     ...
     argN,
+    argN+1,
 )

```

Without the comma or with a closing parens, the diff would be three lines

``` diff
 foo(
     ...
-    argN)
+    argN,
+    argN+1)


```

**Pull request merged**